dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/programrule/executor/event/AssignDataValueExecutor.java

```diff
@@ -90,7 +90,11 @@ public Optional<ProgramRuleIssue> executeRuleAction(TrackerBundle bundle, Event
         || isEqual(value, payloadDataValue.getValue(), dataElement.getValueType())) {
       addOrOverwriteDataValue(event, bundle, dataElement, payloadDataValue);
       return Optional.of(
-          warning(ruleUid, ValidationCode.E1308, dataElementUid.getValue(), event.getEvent()));
+          warning(
```

Comment:
These methods take a vararg for the validation message right? Otherwise they should also accept a UID

Reply:
Yes, they accept Strings.
Anyway, I am going to review our report and programrule package when all the entities are fixed.